tests/presses/test_head_compression.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-FileCopyrightText: Copyright (c) 1993-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+import torch
+from transformers import DynamicCache
+
+from kvpress import AdaKVPress, CriticalAdaKVPress, KnormPress
+from tests.fixtures import unit_test_model  # noqa: F401
+
+
+def compute_masked_percentage(module, batch_size, num_key_value_heads, seq_len):
+    """
+    Compute the percentage of masked indices from module.masked_key_indices.
+    """
+    if module.masked_key_indices is None:
+        return 0.0
+
+    batch_indices, head_indices, seq_indices = module.masked_key_indices
+    num_masked = len(batch_indices)
+    total_positions = batch_size * num_key_value_heads * seq_len
+    masked_percentage = num_masked / total_positions
+    return masked_percentage
+
+
+@pytest.mark.parametrize(
+    "wrapper_press",
+    [AdaKVPress, CriticalAdaKVPress],
+)
+def test_head_compression(unit_test_model, wrapper_press):  # noqa: F811
+    for compression_ratio in [0.2, 0.4, 0.6, 0.8]:
```

Comment:
`@pytest.mark.parametrize` could also be used for `compression_ratio`. Not crucial, but it would pytest allow for more granular reports (e.g. `AdaKVPress` fails at 0.4 `compression_ratio` ).